lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocations.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestLocations
+{
+    private static Stream<Arguments> locations()
+    {
+        return Stream.of(
+                Arguments.of("test_dir", "", "test_dir/"),
+                Arguments.of("", "test_file.txt", "/test_file.txt"),
+                Arguments.of("test_dir", "test_file.txt", "test_dir/test_file.txt"),
+                Arguments.of("test_dir/", "test_file.txt", "test_dir/test_file.txt"),
+                Arguments.of("test_dir", "test_dir2/", "test_dir/test_dir2/"),
+                Arguments.of("test_dir/", "test_dir2/", "test_dir/test_dir2/"),
```

Comment:
How about testing with leading `/`? like "/test_dir"

Reply:
Assuming `Locations` can be used with local filesystem.

Reply:
test added